### PR TITLE
use struct iovec for grouping buffer and length

### PR DESCRIPTION
**Pull request by tmakatos**
This make it tidier and easier to pass to function the buffer and
length, instead of passing the whole msg.

Signed-off-by: Thanos Makatos <thanos.makatos@nutanix.com>

### CODE REVIEW

**Comment by jlevon** [comment]
This change (and your suggested TODO) look like a nice cleanup on their own, but I don't like the idea of "only passing the iovecs" at all. Introducing the msg struct made the request handling code *much* easier to understand. Why do you want to pass just iovecs around?

**Comment by tmakatos** [comment]
> I don't like the idea of "only passing the iovecs" at all. Introducing the msg struct made the request handling code _much_ easier to understand. Why do you want to pass just iovecs around?

The reason I suggested this is because I've added a new function in migration v2 where I only need the out iovec, and I also checked that in other, existing places we only use a subset of the message. I didn't think of it from the angle you state, so we can leave as it is.

File: lib/private.h [hunk 1]
```diff
@@ -60,28 +60,22 @@
...
-    int *out_fds;
-    size_t nr_out_fds;
-
-    void *out_data;
-    size_t out_size;
+    struct {
+        int *fds;
+        size_t nr_fds;
+        struct iovec iov;
+    } in, out;
```
Comment (jlevon) [inline lib/private.h:78]: MIND BLOWN. I had no idea this was allowed!

**Review by jlevon — APPROVED** [review]
LGTM thanks

**Pull request merged**